### PR TITLE
Check configs in a XML file

### PR DESCRIPTION
**Pull request by rsanda**
Hi,

I am trying to test the contents of a XML file on a remote host located at `/etc/hadoop/conf/hdfs-site.xml`:
```
<?xml version="1.0"?>
<?xml-stylesheet type="text/xsl" href="configuration.xsl"?>
<configuration>
  <property>
        <name>dfs.nameservices</name>
        <value>test1</value>
  </property>
</configuration>
```

With the changes I made, Serverspec checks localhost not the remote host. How can I make it to check the remote host?
Usage:
```
describe 'check hdfs' do
  context hdfs('dfs.nameservices') do
    its(:value) { should eq "test1" }
  end
end
``` 

Thank you.

PS: This uses `nokogiri` to parse XML file.

### CODE REVIEW

File: lib/serverspec/type/hdfs.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+require 'nokogiri'
+
+module Serverspec::Type
+  class Hdfs < Base
+    def value
+      regx = /#{@name}/
+      doc = ::Nokogiri::XML( open('/etc/hadoop/conf/hdfs-site.xml') )
```
Comment (mizzy) [inline lib/serverspec/type/hdfs.rb]:
```ruby
doc = ::Nokogiri::XML(@runner.get_file_content('/etc/hadoop/conf/hdfs-site.xml'))
```

File: lib/serverspec/type/hdfs.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+require 'nokogiri'
```
Comment (mizzy) [inline lib/serverspec/type/hdfs.rb]: Nokogori is too heavy to install for the users which do not need use this type.
So please change like this.

```ruby
begin
  require 'nokogiri'
rescue LoadError
  fail "nokogiri is not available. Try installing it."
end
```

File: lib/serverspec/helper/type.rb [hunk 3]
```diff
@@ -10,7 +10,7 @@ module Type
         windows_feature windows_hot_fix windows_registry_key
         windows_scheduled_task zfs docker_base docker_image
         docker_container x509_certificate x509_private_key
-        linux_audit_system
+        linux_audit_system hdfs
```
Comment (mizzy) [inline lib/serverspec/helper/type.rb]: It would be better `hdfs_config` instead of `hdfs`.

File: lib/serverspec/type/hdfs_config.rb [hunk 4]
```diff
@@ -0,0 +1,22 @@
+begin
+  require 'nokogiri'
+rescue LoadError
+  fail "nokogiri is not available. Try installing it."
+end
```
Comment (mizzy) [inline lib/serverspec/type/hdfs_config.rb]: Please put these lines in `HdfsConfig` class like this.

```ruby
module Serverspec::Type
  class HdfsConfig < Base
    def initialize(name=nill)
      super
      
      begin
        require 'nokogiri'
      rescue LoadError
        fail "nokogiri is not available. Try installing it."
      end
    end
```

Comment (rsanda) [inline lib/serverspec/type/hdfs_config.rb]: Hi

Thank you very much for the feedback. I will update the code.

Currently, the XML file name is fixed. Is there any efficient way to make this applicable to any XML file?

Thank you.

Comment (mizzy) [inline lib/serverspec/type/hdfs_config.rb]: How about this like #548 ?

```ruby
describe hdfs_config('dfs.nameservices', :file => '/path/to/file') do
    its(:value) { should eq "test1" }
  end
```

You should rebase your branch on latest master to implement this.

**Comment by mizzy** [comment]
Commits history is strange. You must rebase your branch on master. Don't merge master to your branch.

I will not merge this if you don't clean your history.

If you don't know how to clean, close this pull request and open new one.

**Comment by rsanda** [comment]
Hi Mizzy,

Thank you for the advice. I changed the as u suggested, but it fails to read the file. The error is,
```
Failure/Error: its(:value) { should eq "test1" }
       
       expected: "test1"
            got: 0
       
       (compared using ==)
       /bin/sh -c cat\ \'\'\ 2\>\ /dev/null\ \|\|\ echo\ -n
```
I am not familiar with Ruby. Do you have any idea about this error?

thank you.

**Pull request closed without merging**